lms/views/api/blackboard/files.py

```diff
@@ -71,11 +64,12 @@ def via_url(self):
         course = self.request.find_service(name="course").get_by_context_id(
             course_id, raise_on_missing=True
         )
-
-        document_url = self.request.params["document_url"]
-        file_id = course.get_mapped_file_id(
-            DOCUMENT_URL_REGEX.search(document_url)["file_id"]
+        document_url_file_id = (
+            self.request.find_service(DocumentService)
+            .get_document_url_parts(self.request.params["document_url"])
+            .file_id
```

Comment:
If `get_document_url_parts` can return `None`, would calling `.file_id` on it potentially crash at runtime? Or how does that work in Python?

Reply:
Yes it will crash at runtime. That means the request will fail and we'll get a sentry exception with the details (and of course the user won't be able to see the file).

The previous version has the same behavior, the regexp will fail to match and will return None. In practice that's not a problem because the URL that gets sent here is set by the backend in the first place, if it's not parse-able it's an issue, not normal behaviour.


In any case I added an explicit check and raise to make this all more clear :+1:

Reply:
If this is how we usually handle it, I'm ok with it. I was not implying you should add the explicit check 😅 

Similar logic in the frontend would not be allowed by our linter, that's why I asked.